content/blog/2022/07/verify-signatures-with-gha-and-prefix.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Verify image signatures with GithubActions and KeylessPrefix 
+authors:
+- Raul Cabello Martin 
+date: 2022-07-14
+---
+
+With the latest Kubewarden release v1.1.0 and the latest release of
+the [verify-image-signatures](https://github.com/kubewarden/verify-image-signatures) policy, it's now possible to use
+`GithubActions` or `KeylessPrefix` for verifying images. Read our
+previous [blog post](https://www.kubewarden.io/blog/2022/05/verifying-image-signatures/) if you want to learn more about
+how to verify container images with Sigstore using Kubewarden.
+
+## Let's see it in action!
+
+We want to verify the image `ghcr.io/raulcabello/app-example` which was signed inside a GitHub action. Let's run the
+following command to see the issuer and subject:
+
+```
+COSIGN_EXPERIMENTAL=1 cosign verify ghcr.io/raulcabello/app-example:v0.4.0
+```
+
+The output is:
+
+```json 
+{
+  ....
+  "Issuer": "https://token.actions.githubusercontent.com",
+  "Subject": "https://github.com/raulcabello/app-example/.github/workflows/ci.yml@refs/tags/v0.4.0"
+}
+```
```

Comment:
I don't feel it makes sense to talk about this output. Let's be honest, cosign is not telling you the relevant bits that are used by us when doing a GitHub verification. I'm talking about the x509 extensions that point to the GitHub owner and repository.

I would personally remove all this snippet and just say something like

> We want to verify the image `ghcr.io/raulcabello/app-example` which was built and signed inside a GitHub action
> using [this GitHub Action](https://github.com/raulcabello/app-example/blob/master/.github/workflows/ci.yml).
> 
> Out of the box, GitHub Actions have a specially crafted environment that makes Sigstore keyless signing work in
> a non-interactive way.
> The signatures produced in this way contain unique secure information that allow us to identify the GitHub owner
> (be it an individual or an organization) and the GitHub repository inside of which the GitHub Action has been
> executed.
>
> Starting from today, Kubewarden provides a convenient way to check signatures produced by GitHub actions.

And then you show the GitHub action example.

Reply:
done in https://github.com/kubewarden/kubewarden.io/pull/116/commits/80a27aea1eb710fffac2671ce17859f351b328c8